### PR TITLE
Add `SetNonNullable` type

### LINKED ISSUE #144: [Proposal] SetNonNullable
I honestly thought `SetRequired` did this. example of expected behavior

```ts
type Optional<T> = T | undefined | null;
type Foo = {
   bar: Optional<string>;
   foo?: string | null;
   baz?: string;
};

type NonNullableFoo = SetNonNullable<Foo, 'bar' | 'foo'>;

// would be equivalent to

type ManualNonNullableFoo = {
   bar: string;
   foo: string;
   baz?: string;
};
```

### PR DESCRIPTION
**Pull request by Smtih**
Often, you will want to extract code the checks or asserts that specific fields of an object are present, this makes typing this much easier.

```typescript
function assertHasName(person: Person): asserts person is SetNonNullable<Person, "name"> {
  if(person.name == null) throw new Error("Person has no name");
}
```

Fixes #144

### CODE REVIEW

**Comment by sindresorhus** [comment]
> I wonder if we could just overwrite the built-in NonNullable version with a more powerful one that also accepts which keys to apply to, as discussed in #171.

Any thoughts on this?

**Comment by Smtih** [comment]
While we could do this, I'm unsure what the benefit is.

I think it would be pretty unexpected for a type named non-nullable to do this, especially as it's already a built in type Does it also make the entire object non-nullable as well as the keys?

I think having the word 'Set' makes it clear how this type operates.

File: source/set-non-nullable.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import {Except} from './except';
+import {Simplify} from './simplify';
+
+/**
+Create a type removes nullish (optional, undefined and null) types from the given keys. The remaining keys are kept as is.
+
```
Comment (sindresorhus) [inline source/set-non-nullable.d.ts:6]: Can you mention how it related to `SetRequired` and the inverse in the `SetRequired` doc comment?

File: source/set-non-nullable.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import {Except} from './except';
+import {Simplify} from './simplify';
+
+/**
+Create a type removes nullish (optional, undefined and null) types from the given keys. The remaining keys are kept as is.
```
Comment (sindresorhus) [inline source/set-non-nullable.d.ts:5]:
```suggestion
Create a type that removes nullish (optional, `undefined`, and `null`) types from the given keys. The remaining keys are kept as is.
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Please follow: https://github.com/sindresorhus/type-fest/blob/main/.github/contributing.md

**Comment by sindresorhus** [comment]
Alright, let's go with `Set` like you have it.

**Comment by sindresorhus** [comment]
Friendly bump :)

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/type-fest/issues/228

**Pull request closed without merging**